### ISSUE TITLE
dmg file size

### ISSUE THREAD

**Issue by uwemock** [perdian/macosappbundler-maven-plugin#34]
I have always wondered why the resulting dmg file is so large. My impression is that the bundler uses a minimal JAR (2.5 MB in my case) and bundles all the referenced libraries with that JAR. As libraries have grown, I get a 42 MB dmg file now. The corresponding deb file and the Windows package are some 17 MB large.

So, could you add an option to use the shaded JAR instead of adding each and any file included in some referenced library?

**Comment by perdian** [comment]
I'm not sure if I get your request correctly, so let me try to summarize what I got so far:

If we take the example project at https://github.com/perdian/macosappbundler-maven-plugin/tree/master/example-classpath and perform the bundling process, what we get is the `.app` bundle:

```
$ mvn clean package -Pmacos-bundle
...
$ tree tree target/macOS\ app\ bundler\ Maven\ plugin\ -\ Example\ application.app

target/macOS app bundler Maven plugin - Example application.app
└── Contents
    ├── Info.plist
    ├── Java
    │   └── classpath
    │       ├── de
    │       │   └── perdian
    │       │       └── maven
    │       │           └── plugins
    │       │               └── macosappbundler-maven-plugin-example-classpath
    │       │                   └── 1.0.0-SNAPSHOT
    │       │                       └── macosappbundler-maven-plugin-example-classpath-1.0.0-SNAPSHOT.jar
    │       └── org
    │           └── openjfx
    │               ├── javafx-base
    │               │   └── 11.0.2
    │               │       ├── javafx-base-11.0.2-mac.jar
    │               │       └── javafx-base-11.0.2.jar
    │               ├── javafx-controls
    │               │   └── 11.0.2
    │               │       ├── javafx-controls-11.0.2-mac.jar
    │               │       └── javafx-controls-11.0.2.jar
    │               └── javafx-graphics
    │                   └── 11.0.2
    │                       ├── javafx-graphics-11.0.2-mac.jar
    │                       └── javafx-graphics-11.0.2.jar
    ├── MacOS
    │   └── JavaLauncher
    └── Resources
        └── META-INF
            └── example.css
```

The dependencies (`javafx-base`, `javafx-controls` and `javafx-graphics`) and part of the generated bundle.

If I understand you correctly you want an option to **exclude** them from the resulting `.app`, so that a `tree` would show the following output:

```
$ tree tree target/macOS\ app\ bundler\ Maven\ plugin\ -\ Example\ application.app

target/macOS app bundler Maven plugin - Example application.app
└── Contents
    ├── Info.plist
    ├── Java
    │   └── classpath
    │       └── de
    │           └── perdian
    │               └── maven
    │                   └── plugins
    │                       └── macosappbundler-maven-plugin-example-classpath
    │                           └── 1.0.0-SNAPSHOT
    │                               └── macosappbundler-maven-plugin-example-classpath-1.0.0-SNAPSHOT.jar
    ├── MacOS
    │   └── JavaLauncher
    └── Resources
        └── META-INF
            └── example.css
```

Is that correct?

In that case the application itself wouldn't run correctly as essential dependencies are missing.

**Comment by uwemock** [comment]
These dependencies are included in the shaded JAR which is minimized by the `<minimizeJar>true</minimizeJar>` option. It contains all necessary classes, but those parts from the libraries that are never used are left out.

**Comment by perdian** [comment]
Implemented via https://github.com/perdian/macosappbundler-maven-plugin/commit/541844e9e9bc44e2d661aeec7078e0ee1d410335 and scheduled to be released as `1.17.0`.

Maven Central seems to have some issues at the moment, which prevents me from uploading the new release. Will do that as soon as possible.

**Comment by perdian** [comment]
Issues seem resolved, release pushed to Maven Central as `1.17.0`

**Comment by uwemock** [comment]
Thank you, this works as expected. dmg file has shrunk to less than half its prior size.
